bgp/tests/local_tests/route_type_test.go

```diff
@@ -0,0 +1,132 @@
+				t.Fatalf("Cannot append new BGP policy statement: %v", err)
+			}
+			stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(prefixSetName)
+			stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.PolicyTypes_MatchSetOptionsRestrictedType_ANY)
+
+			switch i {
+			case 0, 3:
+			case 1, 4:
+				stmt.GetOrCreateConditions().GetOrCreateBgpConditions().SetRouteType(oc.BgpConditions_RouteType_INTERNAL)
+			case 2, 5:
+				stmt.GetOrCreateConditions().GetOrCreateBgpConditions().SetRouteType(oc.BgpConditions_RouteType_EXTERNAL)
+			default:
+				t.Fatalf("BGP set policy not specified for test case %d", i)
+			}
+			stmt.GetOrCreateActions().SetPolicyResult(oc.RoutingPolicy_PolicyResultType_REJECT_ROUTE)
+		}
+		// Install policy
+		Replace(t, dut2, ocpath.Root().RoutingPolicy().PolicyDefinition(policyName).Config(), &oc.RoutingPolicy_PolicyDefinition{Name: ygot.String(policyName), Statement: policy})
+		Replace(t, dut2, bgp.BGPPath.Neighbor(dut1.RouterID).ApplyPolicy().ImportPolicy().Config(), []string{policyName})
+		Replace(t, dut2, bgp.BGPPath.Neighbor(dut5.RouterID).ApplyPolicy().ImportPolicy().Config(), []string{policyName})
+		Await(t, dut2, bgp.BGPPath.Neighbor(dut1.RouterID).ApplyPolicy().ImportPolicy().State(), []string{policyName})
+		Await(t, dut2, bgp.BGPPath.Neighbor(dut5.RouterID).ApplyPolicy().ImportPolicy().State(), []string{policyName})
+	}
+
+	test := func() {
+		testName := "test-route-type"
+		t.Run(testName, func(t *testing.T) {
+			testPolicy(t, &PolicyTestCase{
+				description:         "route-type",
+				skipValidateAttrSet: true,
+				dut1IsEBGP:          true,
+				routeTests: []*policytest.RouteTestCase{{
+					Description: "not set",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[0],
+					},
+					ExpectedResult: policytest.RouteDiscarded,
+				}, {
+					Description: "internal",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[1],
+					},
+					ExpectedResult: policytest.RouteAccepted,
+				}, {
+					Description: "external",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[2],
+					},
+					ExpectedResult: policytest.RouteDiscarded,
+				}},
+				alternatePathRouteTests: []*policytest.RouteTestCase{{
+					Description: "not set",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[3],
+					},
+					ExpectedResult: policytest.RouteDiscarded,
+				}, {
+					Description: "internal",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[4],
+					},
+					ExpectedResult: policytest.RouteDiscarded,
+				}, {
+					Description: "external",
+					Input: policytest.TestRoute{
+						ReachPrefix: routeUnderTestList[5],
+					},
+					ExpectedResult: policytest.RouteAccepted,
+				}},
+				installPolicies: func(t *testing.T, dut1, dut2, dut3, dut4, dut5 *Device) {
+					installRejectPolicy(t, dut1, dut2, dut3, dut4, dut5)
+				},
+			})
+		})
+	}
+
+	test()
```

Comment:
why make this a var and call immediately after?

Reply:
Some copy-pasting artifacts. Removed.